tests/network/wireguard.pm

```diff
@@ -0,0 +1,85 @@
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: Connect two machines using wireguard VPN
+#  Run iperf3 speed test inside the tunnel for verification
+# Maintainer: Pavel Dostál <pdostal@suse.cz>
+
+use base 'consoletest';
+use testapi;
+use strict;
+use warnings;
+use utils;
+use lockapi;
+use version_utils;
+use mmapi;
+
+sub run {
+    select_console 'root-console';
+    barrier_wait 'SETUP_DONE';
+
+    my ($vpn_local, $vpn_remote, $remote) = '';
+    if (get_var('HOSTNAME', '') =~ /server/) {
+        $vpn_local  = '192.168.2.1';
+        $vpn_remote = '192.168.2.2';
+        $remote     = '10.0.2.102';
+    } else {
+        $vpn_local  = '192.168.2.2';
+        $vpn_remote = '192.168.2.1';
+        $remote     = '10.0.2.101';
+    }
+
+    if (is_tumbleweed()) {
+        zypper_call 'ar -efG https://download.opensuse.org/repositories/network:/vpn:/wireguard/openSUSE_Tumbleweed/network:vpn:wireguard.repo';
+    } elsif (is_leap('=15.2')) {
+        zypper_call 'ar -efG https://download.opensuse.org/repositories/network:/vpn:/wireguard/openSUSE_Leap_15.2/network:vpn:wireguard.repo';
+    } elsif (is_leap('=15.1')) {
+        zypper_call 'ar -efG https://download.opensuse.org/repositories/network:/vpn:/wireguard/openSUSE_Leap_15.1/network:vpn:wireguard.repo';
+    } else {
+        die 'This OS is not supported';
+    }
+    zypper_call 'se wireguard';
+    zypper_call 'in wireguard-kmp-default wireguard-tools';
+
+    assert_script_run 'which wg';
+    assert_script_run 'umask 077';
+    assert_script_run 'wg genkey > ./private';
+    assert_script_run 'cat ./private';
+    assert_script_run 'wg pubkey < ./private | tee ./public';
+    exec_and_insert_password("scp -o StrictHostKeyChecking=no ./public root\@$remote:'~/remote'");
+    barrier_wait 'KEY_TRANSFERED';
+    assert_script_run 'cat ./remote';
+
+    assert_script_run 'ip link add dev wg0 type wireguard';
+    assert_script_run "ip address add dev wg0 $vpn_local/24";
+    assert_script_run "wg set wg0 listen-port 51820 private-key ./private";
+    assert_script_run "wg set wg0 peer \$(cat ./remote) allowed-ips $vpn_remote/32 endpoint $remote:51820";
+    assert_script_run 'ip link set up dev wg0';
+
+    assert_script_run 'ip a s wg0';
+    assert_script_run 'wg';
+    barrier_wait 'VPN_ESTABLISHED';
+    assert_script_run "ping -c10 $vpn_remote";
+    zypper_call 'in iperf';
+
+    if (get_var('HOSTNAME', '') =~ /server/) {
+        assert_script_run "iperf3 -B $vpn_local -s -D -p 5001";
+        script_retry 'ss -lptn | grep 5001', delay => 3, retry => 3;
+        mutex_create 'server_ready';
+        wait_for_children;
+    } else {
+        mutex_unlock 'server_ready';
+        script_retry "iperf3 -B $vpn_local -t 30 -c $vpn_remote -p 5001", timeout => 60, delay => 3, retry => 3;
+    }
```

Comment:
When it comes to very odd utilities/binaries which most of us do not use or use in a very limited way, it would be nice to use long options rather than short ones.